src/dune_engine/scheduler.ml

```diff
@@ -434,6 +461,12 @@ end = struct
     ; mutable running_count : int
     }
 
+  let is_running t pid =
```

Comment:
This `is_running` function is suspect because it unlocks the mutex, so on return its result may already be incorrect.
In fact this means that we end up racing a `kill` against a `wait`, which is not good.
In particular, if `wait` finishes before `kill` happens, then `kill` will hit a process id that was "released" by free, which means we may kill a different process that happens to share the pid.